test/integration/query-interface.test.js

```diff
@@ -112,6 +113,114 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
         expect(tableNames).to.include('my_test_table2');
       });
     }
+
+    if (dialect.supports.schemas) {
+      describe('schema support tests', () => {
+        const getSequelizeInstanceWithSchema = async () => {
+          const sequelize = createSequelizeInstance({ schema: 'schema_3' });
+          await sequelize.createSchema('schema_3');
+          await sequelize.queryInterface.createTable('schema_3_table_1', { name: DataTypes.STRING }, {
+            schema: 'schema_3',
+          });
+          await sequelize.queryInterface.createTable('schema_3_table_2', { name: DataTypes.STRING }, {
+            schema: 'schema_3',
+          });
+
+          await sequelize.createSchema('schema_4');
+          await sequelize.queryInterface.createTable('schema_4_table_1', { name: DataTypes.STRING }, {
+            schema: 'schema_4',
+          });
+          await sequelize.queryInterface.createTable('schema_4_table_2', { name: DataTypes.STRING }, {
+            schema: 'schema_4',
+          });
+
+          return { sequelize };
+        };
+
+        async function createTestTablesForSchema(schemaName) {
+          await this.queryInterface.createTable(`${schemaName}_table_1`, { name: DataTypes.STRING }, {
+            schema: schemaName,
+          });
+          await this.queryInterface.createTable(`${schemaName}_table_2`, { name: DataTypes.STRING }, {
+            schema: schemaName,
+          });
+        }
+
+        async function createSchemaAndTables(testSchemas = []) {
+          const baseTestSchemas = ['schema_1', 'schema_2', 'sequelize_test'].concat(testSchemas);
+
+          async function _createSchemaAndTables(schemaName) {
+            await this.sequelize.createSchema(schemaName);
+            await Reflect.apply(createTestTablesForSchema, this, [schemaName]);
```

Comment:
When you move these tests to a new TypeScript suite, you won't need to use `.apply` anymore since we don't use `this` in our new suites anymore